### PR TITLE
feat: added generation rule for tenant_user_id by nanoid

### PR DESCRIPTION
**Pull request by rolin999**
### Description

TenantUserID 支持 NanoID 生成方式

### Checklist

- [ ] 填写 PR 描述及相关 issue (write PR description and related issue)
- [ ] 代码风格检查通过 (code style check passed)
- [ ] PR 中包含单元测试 (include unit test)
- [ ] 单元测试通过 (unit test passed)
- [ ] 本地开发联调环境验证通过 (local development environment verification passed)

### CODE REVIEW

**Review by narasux — CHANGES_REQUESTED** [review]
为什么把 Review 完成的 PR 关掉重开一个？

File: src/bk-user/bkuser/utils/nanoid.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+#
+# We undertake not to change the open source license (MIT license) applicable
+# to the current version of the project delivered to anyone in the future.
+
+from nanoid import generate
+
+
+def generate_nanoid(size: int = 16) -> str:
+    # 字符集由 0-9 的阿拉伯数字与小写字母组成，默认长度为 16
+    return generate(alphabet="0123456789abcdefghijklmnopqrstuvwxyz", size=size)
```
Comment (nannan00) [inline src/bk-user/bkuser/utils/nanoid.py:23]: 1. alphabet  = string.digits + string.ascii_lowercase
2. "字符集由 0-9 的阿拉伯数字与小写字母组成，默认长度为 16" 是函数的注释，并不是下面语句的注释

**Comment by rolin999** [comment]
[https://github.com/TencentBlueKing/bk-user/pull/2017#discussion_r1886380465](url)
[https://github.com/TencentBlueKing/bk-user/pull/2017#discussion_r1886387391](url)
[https://github.com/TencentBlueKing/bk-user/pull/2017#discussion_r1886388875](url)
[https://github.com/TencentBlueKing/bk-user/pull/2017#discussion_r1886393722](url)
[https://github.com/TencentBlueKing/bk-user/pull/2017#discussion_r1886531462](url)
[https://github.com/TencentBlueKing/bk-user/pull/2017#discussion_r1886534287](url)

File: src/bk-user/bkuser/apps/tenant/utils.py [hunk 2]
```diff
@@ -62,17 +63,19 @@ def __init__(self, target_tenant_id: str, data_source: DataSource, prepare_batch
...
         if self.cfg.rule == TenantUserIdRuleEnum.USERNAME:
             return user.username
 
-        return self._reuse_or_generate_uuid(user)
+        return self._reuse_or_generate_id(user, self.cfg.rule)
 
-    def _reuse_or_generate_uuid(self, user: DataSourceUser) -> str:
+    def _reuse_or_generate_id(
+        self, user: DataSourceUser, rule: TenantUserIdRuleEnum = TenantUserIdRuleEnum.NANOID
+    ) -> str:
```
Comment (wklken) [inline src/bk-user/bkuser/apps/tenant/utils.py:78]: 这个函数只能接收 rule= NANOID / UUID4_HEX? 是的话需要做校验，超过范围抛异常（TenantUserIdRuleEnum 范围很大）

File: src/bk-user/bkuser/apps/tenant/utils.py [hunk 3]
```diff
@@ -30,7 +31,7 @@ def is_username_frozen(data_source: DataSource) -> bool:
     """数据源用户名是否不可变更"""
     return (
         TenantUserIDGenerateConfig.objects.filter(data_source=data_source)
-        .exclude(rule=TenantUserIdRuleEnum.UUID4_HEX)
+        .exclude(rule__in=[TenantUserIdRuleEnum.NANOID, TenantUserIdRuleEnum.UUID4_HEX])
```
Comment (wklken) [inline src/bk-user/bkuser/apps/tenant/utils.py:34]: - NIT： _is_username_imutable? 
- 这里是否在 TenantUserIdRuleEnum 声明的后面，跟一个 `MUTABLE_USER_ID_RULES = []`会好一些？ 感觉非常容易漏掉这种地方     @nannan00 一起看下

File: src/bk-user/bkuser/utils/nanoid.py (hunk 1 above)
Comment (rolin999) [reply src/bk-user/bkuser/utils/nanoid.py]: Done

File: src/bk-user/bkuser/apps/tenant/utils.py (hunk 2 above)
Comment (rolin999) [reply src/bk-user/bkuser/apps/tenant/utils.py:78]: Done

File: src/bk-user/bkuser/apps/tenant/utils.py (hunk 3 above)
Comment (rolin999) [reply src/bk-user/bkuser/apps/tenant/utils.py]: Done

**Review by nannan00 — APPROVED** [review]

**Pull request merged**